pkg/config: Support prefix for pprof config (#931)

importhuman opened the pull request:
It seems I may have misunderstood the issue. 

Looking at the 2 configs in the issue description, I thought it meant discovering pprof configs (from somewhere) once the user has specified a particular prefix. However, from [this comment](https://github.com/parca-dev/parca/issues/931#issuecomment-1117506666) I realise it means not having to write a particular string repeatedly while specifying multiple configs. If it is the second case that needs to be addressed, this PR should be good.

Need help understanding relabelling though and what it means in the context of this issue.

brancz commented:
This implementation is pretty much exactly what I expected. What were your thoughts @metalmatze @roidelapluie @kakkoyun?

kakkoyun commented on pkg/config/config_test.go:
```diff
@@ -112,6 +112,84 @@ scrape_configs:
+func TestLoadPrefixConfig(t *testing.T) {
+	prefixYAML := `
+scrape_configs:
+  - job_name: 'conprof'
```
```suggestion
  - job_name: 'parca'
```

kakkoyun commented on pkg/config/config_test.go:
```diff
@@ -112,6 +112,84 @@ scrape_configs:
+      pprof_config:
+        memory:
+          enabled: true
+          path: /conprof/debug/pprof/allocs
```
```suggestion
          path: /parca/debug/pprof/allocs
```

kakkoyun commented on pkg/config/config_test.go:
```diff
@@ -112,6 +112,84 @@ scrape_configs:
+	expected := &Config{
+		ScrapeConfigs: []*ScrapeConfig{
+			{
+				JobName:        "conprof",
```
```suggestion
				JobName:        "parca",
```

roidelapluie commented on pkg/config/config.go:
```diff
@@ -205,6 +206,13 @@ func (c *ScrapeConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	// If path prefix is specified, add to PprofConfig path
+	if unmarshalled.ProfilingConfig.PprofPrefix != "" {
+		for pt := range unmarshalled.ProfilingConfig.PprofConfig {
+			unmarshalled.ProfilingConfig.PprofConfig[pt].Path = unmarshalled.ProfilingConfig.PprofPrefix + unmarshalled.ProfilingConfig.PprofConfig[pt].Path
```
I think we should use `path.Join`

kakkoyun reviewed (commented):
LGTM 💯 

Just a couple of nits

importhuman commented:
I've updated the PR with the suggested changes, and also updated existing test to use `parca` instead of `conprof`.
I'll do some more research on relabelling before starting on it.
